src/common/utils/viewability-manager.js

```diff
@@ -0,0 +1,157 @@
+//@flow
+import {EventManager, Utils} from '@playkit-js/playkit-js';
+
+/**
+ * A service class to observe viewability of elements in the view port.
+ */
+class ViewabilityManager {
+  _observer: window.IntersectionObserver;
+  _targetsObserved: Utils.MultiMap<HTMLElement, _TargetObserveredBinding>;
+  _viewabilityConfig: KPViewabilityConfigObject;
+  _eventManager: EventManager;
+  _visibilityTabChangeEventName: string;
+  _visibilityTabHiddenAttr: string;
+
+  /**
+   * Whether the player browser tab is active or not
+   * @type {boolean}
+   * @private
+   */
+  _isTabVisible: boolean;
+
+  /**
+   * @param {number} viewabilityConfig - the configuration needed to create the manager
+   * @constructor
+   */
+  constructor(viewabilityConfig: KPViewabilityConfigObject = {}) {
+    viewabilityConfig.observedThresholds = viewabilityConfig.observedThresholds || DEFAULT_OBSERVED_THRESHOLDS;
+    viewabilityConfig.playerThreshold =
+      typeof viewabilityConfig.playerThreshold === 'number' ? viewabilityConfig.playerThreshold : DEFAULT_PLAYER_THRESHOLD;
+
+    this._viewabilityConfig = viewabilityConfig;
```

Comment:
nitpicking, but I think it is somewhat implied by `this._config` that it is the ViewabilityManager config, so can be shortened.

Reply:
done